core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java

```diff
@@ -699,7 +701,16 @@ else if (Tables.getTableState(context.getInstance(), table) == TableState.OFFLIN
 
     }
 
-    void addMutations(MutationSet mutationsToSend) {
+    void queueMutations(final MutationSet mutationsToSend) {
+      queueThreadPool.submit(new Runnable() {
```

Comment:
Taking your idea and going further, if we are only using one thread then we can use a timer driven approach and create zero Runnable objects. I'll update the code shortly.